[MUSIC] Compose BGM & SFX (#24)

jeong-hyeonHwang opened the pull request:
@commitcomplete
@Juhwa-Lee1023
@LeeSungNo-ian

### 변경 및 추가 사항
- SoundManager 반복되는 구문 단일 메소드화
- SoundManager BGM Sound Fade In Fade Out 관련 메소드 (changeBGMVolume) 추가
- SoundManager GameOver Sound 재생 관련 사항 추가

Close #24

jeong-hyeonHwang committed:
[MUSIC] Add BGM & SFX Sample

jeong-hyeonHwang committed:
[FEAT] Sound Comment to Active Code

- SoundManager.shared.playBGM()
- SoundManager.shared.playSFX()

jeong-hyeonHwang committed:
Revert "[FEAT] Sound Comment to Active Code"

This reverts commit dc203e791f8c83e87405976705191fd1db2803f3.

jeong-hyeonHwang committed:
[STYLE] Revise Sound File Path

jeong-hyeonHwang committed:
[STYLE] SoundManager Code Comment -> Active Code

jeong-hyeonHwang committed:
[STYLE] Remove Useless Code

jeong-hyeonHwang committed:
[STYLE] Remove Useless Code

jeong-hyeonHwang committed:
[REFACTOR] Add Sound Prepare Method

jeong-hyeonHwang committed:
[FIX] Revise SFX Prepare Method

- One for All

jeong-hyeonHwang committed:
[FEAT] Add GameOver Case Sound Play Code

jeong-hyeonHwang committed:
[STYLE] Revise Property Name

- sfx_eatPlayer -> sfx_EatPlayer
- sfx_gameOverPlayer -> sfx_GameOverPlayer

jeong-hyeonHwang committed:
[FIX] Add Volume Parameter to Sound Prepare Method

jeong-hyeonHwang committed:
[FEAT] Add BGM Volume Change Method

jeong-hyeonHwang committed:
[FEAT] Add duration parameter to BGM Volume Method

jeong-hyeonHwang committed:
[FIX] Revise ViewController's BGMVolume Method

LeeSungNo-ian commented on Bubblia/Bubblia/SoundManager.swift:
```diff
@@ -10,7 +10,8 @@ import AVFoundation
 class SoundManager {
     static var shared = SoundManager()
     private var bgmPlayer = AVAudioPlayer()
-    private var sfxPlayer = AVAudioPlayer()
+    private var sfx_EatPlayer = AVAudioPlayer()
+    private var sfx_GameOverPlayer = AVAudioPlayer()
```
파라미터 앞 `sfx_` 를 붙인 이유가 있으시나요?

jeong-hyeonHwang replied:
AGUAGU 내부에서 사용되는 사운드의 종류가 배경음악(BGM)과 사운드 효과(SFX)로 종류가 나뉘어져 있어, 이를 명시적으로 구분하기 위해 sfx를 붙였습니다.
첫 단어에 공통점을 부각시키고자 sfx를 넣어, 프로퍼티 이름에 언더바까지 있을 필요성은 없는 것 같으니 언더바는 지우겠습니다!
리뷰 감사합니다👍👍👍

jeong-hyeonHwang committed:
[STYLE] Revise Property Name

**SoundManager.swift**
- sfx_EatPlayer -> sfxEatPlayer
sfx_GameOverPlayer -> sfxGameOverPlayer

Juhwa-Lee1023 commented on Bubblia/Bubblia/SoundManager.swift:
```diff
@@ -22,30 +23,54 @@ class SoundManager {
-            print("SFX CAN'T PLAY")
-        }
-        sfxPlayer.play()
+    func pauseBGM() {
+        bgmPlayer.stop()
+    }
+    
+    func changeBGMVolume(volume: Float, duration: CGFloat) {
+        bgmPlayer.setVolume(volume, fadeDuration: duration)
+    }
```
`bgmPlayer`관련 function들 enum으로 묶으면 코드 가독성이 좋아질 것 같습니다!

Juhwa-Lee1023 commented on Bubblia/Bubblia/SoundManager.swift:
```diff
@@ -22,30 +23,54 @@ class SoundManager {
+    func changeBGMVolume(volume: Float, duration: CGFloat) {
+        bgmPlayer.setVolume(volume, fadeDuration: duration)
+    }
+    
+    func prepareSFX_Eat() {
+        let index = Int.random(in: 1...2)
+        sfxEatPlayer = playerPrepare(source: "AGUAGU_SFX\(index)", loop: false, volume: 1.0)
+    }
+    
+    func playSFX_Eat() {
+        prepareSFX_Eat()
+        sfxEatPlayer.play()
+    }
+    
+    func playSFX_GameOver() {
+        sfxGameOverPlayer.play()
```
이거도 enum으로!

Juhwa-Lee1023 reviewed (commented):
사운드관련 메소드들 많이 배우고 갑니다!

LeeSungNo-ian commented:
고생하셨습니다!
